encoders/firebase-decoders-json/src/main/java/com/google/firebase/decoders/TypeToken.java

```diff
@@ -118,7 +110,7 @@ private TypeToken() {}
 
     private ClassToken(Class<T> token) {
       this.rawType = token;
-      this.typeArguments = null;
+      this.typeArguments = new TypeTokenContainer(null);
```

Comment:
suggestion: Since `TypeTokenContainer` is immutable, let's have a singleton `TypeTokenContainer .EMPTY` instance that can be shared by all non-generic classes as opposed to allocating one each time.